### DISCUSSION TITLE
Eigenstrain derivative for ComputeMeanThermalExpansionFunctionEigenstrain

### DISCUSSION THREAD

**Discussion by chakra34** [idaholab/moose#25620] [Q&A Modules: Solid mechanics]
Hi,

I was going through the source code for ```ComputeMeanThermalExpansionFunctionEigenstrain```  and it seems that the eigenstrain derivative is calculated as:

![image](https://github.com/idaholab/moose/assets/22983992/2a9b4291-8cf1-4499-85f9-55d46483c2c3)

But if I look at the equation of thermal strain which is:
```Eig_strain(T) = alpha_mean(T,T_ref) * (T - T_ref)```, 

then shouldn't the derivative ```d{Eig_strain(T)}/dT``` be:
```
d{Eig_strain(T)}/dT = (d{alpha_mean(T,T_ref)}/dT) * (T - T_ref) + alpha_mean(T,T_ref)*dT/dT = (d{alpha_mean(T,T_ref)}/dT) * (T - T_ref) + alpha_mean(T,T_ref)

````
instead of the coded

```
d{Eig_strain(T)}/dT = (d{alpha_mean(T,T_ref)}/dT) * dT/dT = (d{alpha_mean(T,T_ref)}/dT)

```
Am I missing something ?

Thanks,
Best,

Aritra

**Comment by GiudGiud** [comment]
@bwspenc @dschwen worked on that code
